### PR TITLE
connect database, add migrations and seeds

### PR DESCRIPTION
**Pull request by dolphin323**

### CODE REVIEW

File: backend/src/data/seeders/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { createConnection } from 'typeorm';
+import UserProfileSeeder from './userProfileSeeder';
+
+createConnection().then(async () => {
+  await UserProfileSeeder.execute();
+}).catch(e => {
```
Comment (yevhenii-moroziuk) [inline backend/src/data/seeders/index.ts]: Use log from logger

File: backend/src/server.ts [hunk 2]
```diff
@@ -11,3 +12,4 @@ const app = new MicroMq({
 });
 
 initRoutes(app).start();
+createConnection().catch(e => console.log(e));
```
Comment (yevhenii-moroziuk) [inline backend/src/server.ts]: createConnection().catch(log);

File: backend/src/services/user/index.ts [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+import { UserRepository } from '../../data/repositories';
 import UserService from './user';
 
-const userService = new UserService(
-  userRepository
-);
+// eslint-disable-next-line import/no-mutable-exports
+let userService: UserService = null;
+createConnection().then(() => {
+  userService = new UserService(
```
Comment (yevhenii-moroziuk) [inline backend/src/services/user/index.ts]: Init UserService in Services/user/index.

const userService = new UserService(getCustomRepository(UserRepository))

File: gateway/src/helpers/arrayHelper.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const asyncForEach = async <T>(fn: (x: T) => Promise<void>, list: readonly T[]) => {
```
Comment (yevhenii-moroziuk) [inline gateway/src/helpers/arrayHelper.ts]: Move asyncForEach to shared

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (yevhenii-moroziuk) [inline .vscode/settings.json]: Do not change vscode settings. We will add prettier later

File: backend/src/api/routes/user.ts [hunk 6]
```diff
@@ -5,6 +5,6 @@ import { wrap } from '../../helpers';
 
 const init = ({ userService }: Services, path: string, app: MicroMq) => app
   .get(`${path}`, wrap(() => userService.getAll()))
-  .get(`${path}/:id`, wrap<Empty, User, { id: string }, Empty>((req) => userService.getById(req.params.id)));
+  .get(`${path}/:id`, wrap<Empty, User[], { id: string }, Empty>((req) => userService.getById(req.params.id)));
```
Comment (yevhenii-moroziuk) [inline backend/src/api/routes/user.ts]: Why getById returns a list of users?

File: backend/src/data/repositories/user.ts [hunk 7]
```diff
@@ -1,35 +1,13 @@
...
+@EntityRepository(UserProfile)
+export class UserRepository extends Repository<UserProfile> {
   public getAll() {
-    return Promise.resolve(this.#users);
+    return this.find();
   }
 
   public getById(id: string) {
-    return Promise.resolve(this.#users.find(it => it.id === Number(id)));
+    return this.find({ id });
```
Comment (yevhenii-moroziuk) [inline backend/src/data/repositories/user.ts]: Use findOne instead

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

File: gateway/src/helpers/arrayHelper.ts (hunk 4 above)
Comment (mtseluiko) [reply gateway/src/helpers/arrayHelper.ts]: Try not to use eslint-disable as much as possible 
`return await list.reduce(async (listPromise, item) => {
	await listPromise;
	await fn(item);
}, null);`

**Review by mtseluiko — CHANGES_REQUESTED** [review]

**Review by yevhenii-moroziuk — APPROVED** [review]

**Pull request merged**